youtube_comedy_slam/youtube_comments_api.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+import httplib2
+import os
+import sys
+import pprint
+import csv
+
+from apiclient.discovery import build
+from apiclient.errors import HttpError
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.file import Storage
+from oauth2client.tools import argparser, run_flow
+
+# CLIENT_SECRETS_FILE = "client_secrets.json"
+DEVELOPER_KEY = "AIzaSyAzV-ojzFgf3NSi9V8v9f_Ew2xQcRHrF84"
+YOUTUBE_API_SERVICE_NAME = "youtube"
+YOUTUBE_API_VERSION = "v3"
+
+youtube = build(YOUTUBE_API_SERVICE_NAME, YOUTUBE_API_VERSION, developerKey=DEVELOPER_KEY)
+
+
+
+
+def get_comment_threads(youtube, video_id, csvfile):
+    """
+    Our purpose is to download comments and append them to a csv with the following format:
+    videoId,textDisplay, canReply, totalReplyCount,etag,id,authorChannelId,authorDisplayName,likeCount,publishedAt,updatedAt,viewerRating
+
+    """
+    results = youtube.commentThreads().list(
+        part="snippet,replies",
+        textFormat="plainText",
+        maxResults=100,
+        videoId=video_id
+    ).execute()
+
+
+    # pp = pprint.PrettyPrinter(indent=4)
+    # pp.pprint(results)
+
+    write_header = not os.path.isfile(csvfile)
+
+    ## Setup csv file
+    fieldnames = ['videoId', 'textDisplay', 'canReply', 'totalReplyCount', 'kind', 'etag', 'id', 'authorChannelId',
+                  'authorDisplayName', 'likeCount', 'publishedAt', 'updatedAt', 'viewerRating']
+    with open(csvfile, 'a') as fcsv:
+        writer = csv.DictWriter(fcsv, fieldnames=fieldnames)
+        if write_header:
+            writer.writeheader()
+
+        def get_comment_page(results, ignoreFirst = False):
+            first = True
+            for item in results["items"]:
+                if first:
+                    first = False
+                    continue
+                snippet = item["snippet"]
+                comment = snippet["topLevelComment"]["snippet"]
+                videoId = comment["videoId"]
+                textDisplay = comment["textDisplay"]
+                canReply = snippet["canReply"]
+                totalReplyCount = snippet["totalReplyCount"]
+                kind = item["kind"]
+                etag = snippet["topLevelComment"]["etag"]
+                id = snippet["topLevelComment"]["id"]
+                authorChannelId = comment["authorChannelId"] if "authorChannelId" in comment else ""
```

Comment:
More common in python is:
`authorChannelId = comment.get("authorChannelId", "")`